src/Integration/LocalServices/SolutionWorkspaceService.cs

```diff
@@ -65,17 +65,18 @@ public IEnumerable<string> ListFiles()
         }
 
         [ExcludeFromCodeCoverage]
-        private IEnumerable<string> GetAllFilesInSolution(IVsSolution solution)
+        private ICollection<string> GetAllFilesInSolution(IVsSolution solution)
         {
-            IEnumerable<string> result = null;
+            ICollection<string> result = null;
             threadHandling.RunOnUIThread(() => result = GetLoadedProjects(solution)
                 .SelectMany(AllItemsInProject)
                 .Where(x => x != null)
                 .Where(x => x.Contains("\\"))
                 .Where(x => !x.EndsWith("\\"))
                 .Where(x => !x.Contains("\\.nuget\\"))
-                .Where(x => !x.Contains("\\node_modules\\"))); // move filtering closer to path extraction to avoid processing unnecessary items)
-
+                .Where(x => !x.Contains("\\node_modules\\"))
+                .ToHashSet(StringComparer.InvariantCultureIgnoreCase)); // move filtering closer to path extraction to avoid processing unnecessary items)
```

Comment:
This will be better as `.Distinct(...).ToArray` once this returns an array

Reply:
See IReadOnlyCollection change